android/src/main/java/com/thebluealliance/androidclient/binders/TeamInfoBinder.java

```diff
@@ -165,6 +178,55 @@ private void bindNoDataView() {
         }
     }
 
+    private void toggleFullTeamNameExpanded() {
+        int currentMaxLines = teamFullName.getMaxLines();
+        if (currentMaxLines == TEAM_FULL_NAME_COLLAPSED_MAX_LINES) {
+            // The text view is collapsed, expand it
+
+            final int height = teamFullName.getMeasuredHeight();
+
+            teamFullName.setMaxLines(Integer.MAX_VALUE);
+            teamFullName.measure(
+                    View.MeasureSpec.makeMeasureSpec(teamFullName.getMeasuredWidth(), View.MeasureSpec.AT_MOST),
+                    View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED));
+            final int newHeight = teamFullName.getMeasuredHeight();
+
+            ObjectAnimator animation = ObjectAnimator.ofInt(teamFullName, "height", height, newHeight);
+            animation.setDuration(500);
+            animation.addListener(new AnimatorListenerAdapter() {
+                @Override public void onAnimationEnd(Animator animation) {
+                    teamFullName.setMaxLines(Integer.MAX_VALUE);
+                    teamFullName.setMinHeight(0);
+                }
+            });
+            animation.start();
+        } else {
```

Comment:
Yeah, in retrospect I could have reused code a bit here. The two clauses were originally very different, didn't realize how much they converged as I iterated :P